src/Auth/IAuthorizationResolver.cs

```diff
@@ -33,18 +33,18 @@ public interface IAuthorizationResolver
         /// <param name="roleName">Role defined in client role header</param>
         /// <param name="operation">Operation type: Create, Read, Update, Delete</param>
         /// <returns>True, if a matching permission entry is found.</returns>
-        public bool AreRoleAndOperationDefinedForEntity(string entityName, string roleName, Operation operation);
+        public bool AreRoleAndOperationDefinedForEntity(string entityName, string roleName, EntityActionOperation operation);
 
         /// <summary>
         /// Any columns referenced in a request's headers, URL(filter/orderby/routes), and/or body
         /// are compared against the inclued/excluded column permission defined for the entityName->roleName->operation
         /// </summary>
-        /// <param name="entityName">Entity from request</param>
+        /// <param name="graphQLTypeName">Entity from request</param>
         /// <param name="roleName">Role defined in client role header</param>
         /// <param name="operation">Operation type: Create, Read, Update, Delete</param>
         /// <param name="columns">Compiled list of any column referenced in a request</param>
         /// <returns></returns>
-        public bool AreColumnsAllowedForOperation(string entityName, string roleName, Operation operation, IEnumerable<string> columns);
+        public bool AreColumnsAllowedForOperation(string graphQLTypeName, string roleName, EntityActionOperation operation, IEnumerable<string> columns);
```

Comment:
I renamed this to match what the value is that's provided when the method is called and the callsite for this uses the `NodeName` of the GraphQL type from the inbound query. Since the GraphQL type name isn't always going to be the entity name, since `@model` in CosmosDB schemas can be used to map it, I renamed it so that it was more obvious here.

Admittedly this is really an edge-case, where you have an entity in config that doesn't match the CosmosDB GraphQL type (we test it with `StarAlias` in our tests), but it can happen.

Reply:
we also call into this from the `RestAuthorizationHandler`
![image](https://github.com/Azure/data-api-builder/assets/93220300/c0586345-92d5-4e08-9523-68803ad2e67f)

in this case we are just passing in an entity name rather than a GraphQL type, at least from what I can tell.

Maybe a name that is intuitive for covering both these cases is better to alleviate confusion?

Reply:
Ah, I wonder how I missed that callsite. Ok, I'll rethink how that argument is named.